File: server/internal/models/tax_channel.go
package models

import (
	"time"
)

// 税筹通道扣费类型
const (
	TaxFeeTypePayment    int16 = 1 // 付款扣(充值时扣除)
	TaxFeeTypeWithdrawal int16 = 2 // 出款扣(提现时扣除)
)

// 税筹通道状态
const (
	TaxChannelStatusDisabled int16 = 0 // 禁用
	TaxChannelStatusEnabled  int16 = 1 // 启用
)

// TaxChannel 税筹通道
type TaxChannel struct {
	ID          int64     `json:"id" gorm:"primaryKey"`
	ChannelCode string    `json:"channel_code" gorm:"column:channel_code;uniqueIndex;size:32"`
	ChannelName string    `json:"channel_name" gorm:"column:channel_name;size:100"`
	FeeType     int16     `json:"fee_type" gorm:"column:fee_type"`           // 1=付款扣 2=出款扣
	TaxRate     float64   `json:"tax_rate" gorm:"column:tax_rate;type:decimal(5,4)"` // 税率 如0.09表示9%
	FixedFee    int64     `json:"fixed_fee" gorm:"column:fixed_fee"`         // 固定费用(分)
	ApiURL      string    `json:"api_url" gorm:"column:api_url;size:255"`
	ApiKey      string    `json:"-" gorm:"column:api_key;size:255"`
	ApiSecret   string    `json:"-" gorm:"column:api_secret;size:255"`
	Status      int16     `json:"status" gorm:"column:status"`
	Remark      string    `json:"remark" gorm:"column:remark;size:500"`
	CreatedAt   time.Time `json:"created_at" gorm:"column:created_at"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"column:updated_at"`
}

func (TaxChannel) TableName() string {
	return "tax_channels"
}

// ChannelTaxMapping 支付通道与税筹通道关联
type ChannelTaxMapping struct {
	ID           int64     `json:"id" gorm:"primaryKey"`
	ChannelID    int64     `json:"channel_id" gorm:"column:channel_id"`
	WalletType   int16     `json:"wallet_type" gorm:"column:wallet_type"` // 钱包类型
	TaxChannelID int64     `json:"tax_channel_id" gorm:"column:tax_channel_id"`
	CreatedAt    time.Time `json:"created_at" gorm:"column:created_at"`
	UpdatedAt    time.Time `json:"updated_at" gorm:"column:updated_at"`
}

func (ChannelTaxMapping) TableName() string {
	return "channel_tax_mappings"
}

// GetFeeTypeName 获取扣费类型名称
func GetFeeTypeName(feeType int16) string {
	switch feeType {
	case TaxFeeTypePayment:
		return "付款扣"
	case TaxFeeTypeWithdrawal:
		return "出款扣"
	default:
		return "未知"
	}
}

// GetTaxChannelStatusName 获取状态名称
func GetTaxChannelStatusName(status int16) string {
	switch status {
	case TaxChannelStatusEnabled:
		return "启用"
	case TaxChannelStatusDisabled:
		return "禁用"
	default:
		return "未知"
	}
}

File: web/src/types/taxChannel.ts
// 税筹通道扣费类型
export type TaxFeeType = 1 | 2 // 1=付款扣 2=出款扣

// 税筹通道状态
export type TaxChannelStatus = 0 | 1 // 0=禁用 1=启用

// 税筹通道
export interface TaxChannel {
  id: number
  channel_code: string
  channel_name: string
  fee_type: TaxFeeType
  fee_type_name: string
  tax_rate: number
  tax_rate_percent: number // 百分比显示
  fixed_fee: number
  fixed_fee_yuan: number
  status: TaxChannelStatus
  status_name: string
  remark: string
  created_at: string
  updated_at: string
}

// 创建税筹通道请求
export interface CreateTaxChannelParams {
  channel_code: string
  channel_name: string
  fee_type: TaxFeeType
  tax_rate: number // 小数形式，如0.09
  fixed_fee?: number // 分
  api_url?: string
  api_key?: string
  api_secret?: string
  remark?: string
}

// 更新税筹通道请求
export interface UpdateTaxChannelParams {
  channel_name?: string
  fee_type?: TaxFeeType
  tax_rate?: number
  fixed_fee?: number
  api_url?: string
  api_key?: string
  api_secret?: string
  status?: TaxChannelStatus
  remark?: string
}

// 通道-税筹通道映射
export interface ChannelTaxMapping {
  id: number
  channel_id: number
  wallet_type: number
  wallet_type_name: string
  tax_channel_id: number
  tax_channel_name: string
  tax_rate: number
  fixed_fee: number
  created_at: string
}

// 设置通道税筹映射请求
export interface SetChannelTaxMappingParams {
  channel_id: number
  wallet_type: number // 1=分润 2=服务费 3=奖励 4=充值 5=沉淀
  tax_channel_id: number
}

// 计算税费请求
export interface CalculateTaxParams {
  channel_id: number
  wallet_type: number
  amount: number // 分
}

// 税费计算结果
export interface TaxCalculationResult {
  original_amount: number // 原金额(分)
  tax_rate: number // 税率
  tax_fee: number // 税费(分)
  fixed_fee: number // 固定费用(分)
  total_fee: number // 总费用(分)
  actual_amount: number // 实际到账(分)
  tax_channel_id: number
  tax_channel_name: string
}

// 税筹通道查询参数
export interface TaxChannelQueryParams {
  status?: TaxChannelStatus
}

// 扣费类型配置
export const TAX_FEE_TYPE_CONFIG: Record<TaxFeeType, { label: string; description: string }> = {
  1: { label: '付款扣', description: '充值时扣除税费' },
  2: { label: '出款扣', description: '提现时扣除税费' },
}

// 税筹通道状态配置
export const TAX_CHANNEL_STATUS_CONFIG: Record<TaxChannelStatus, { label: string; color: string }> = {
  0: { label: '禁用', color: '#909399' },
  1: { label: '启用', color: '#67c23a' },
}
